server/src/main/java/org/elasticsearch/index/IndexVersions.java

```diff
@@ -92,6 +92,8 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion ES_VERSION_8_12 = def(8_500_004, Version.LUCENE_9_8_0);
     public static final IndexVersion NORMALIZED_VECTOR_COSINE = def(8_500_005, Version.LUCENE_9_8_0);
     public static final IndexVersion UPGRADE_LUCENE_9_9 = def(8_500_006, Version.LUCENE_9_9_0);
+    public static final IndexVersion ES_VERSION_8_13 = def(8_500_007, Version.LUCENE_9_9_0);
+    public static final IndexVersion UPGRADE_LUCENE_9_9_1 = def(8_500_008, Version.LUCENE_9_9_1);
```

Comment:
@thecoop I am backporting a change to upgrade Lucene 9.9.1, from 8.13 to the currently unreleased 8.12.  As you can see an index version was added in _main_ (currently accumulating changes for 8.13). My understanding is that it we should use the same index version number for the  `UPGRADE_LUCENE_9_9_1` constant (as that in _main_), right?  

Also, the backport of `ES_VERSION_8_13` is not strictly necessary, but I did this to keep the version constants aligned, between 8.12 and _main_. This is the correct procedure, right?

Reply:
Yes, this is correct for the 8.12 branch. I also suggest the following:
- Rename the ES_VERSION_8_13 constant to something else (keep the id)
- Add a new 8.13 constant to main to ensure there's still a new indexversion for 8.13

Reply:
> Yes, this is correct for the 8.12 branch. I also suggest the following:
> 
> * Rename the ES_VERSION_8_13 constant to something else (keep the id)

Agreed. `ES_VERSION_8_13` is not a great name, given where we are now. I'll rename to `ES_VERSION_TRANSIENT`.  This version should be removable at some point in the not too distant future.

> * Add a new 8.13 constant to main to ensure there's still a new indexversion for 8.13

Ah yes, after this change then _main_ will need a new constant to represent that 8.13 is greater than 8.12. The name `ES_VERSION_8_13`seems appropriate here.  I'll get this done.